store/supabase.js

```diff
@@ -0,0 +1,133 @@
+// state
+export const state = () => ({
+  topics: [],
+  members: [],
+  edges: [],
+  nodes: [],
+});
+
+export const actions = {
+  async fetchData({ commit }) {
+    // hit supabase to get topics, members and mappings
+    const { data: topics, error: topicError } = await this.$supabase
+      .from("topics")
+      .select("*");
+    const { data: members, error: memberError } = await this.$supabase
+      .from("members")
+      .select("*");
+    const { data: mapping, error: mappingError } = await this.$supabase
+      .from("members_topics")
+      .select("*");
+
+    if (!topicError) {
+      commit("setTopics", topics);
+    }
+    if (!memberError) {
+      commit("setMembers", members);
+    }
+
+    if (!topicError && !memberError && !mappingError) {
+      const { edges, nodes } = createNodes(members, topics, mapping);
+      commit("setEdges", edges);
+      commit("setNodes", nodes);
+    }
+  },
+};
+
+export const mutations = {
+  setTopics(state, topics) {
+    state.topics = topics;
+  },
+  setMembers(state, members) {
+    state.members = members;
+  },
+  setEdges(state, edges) {
+    state.edges = edges;
+  },
+  setNodes(state, nodes) {
+    state.nodes = nodes;
+  },
+};
+
+function createNodes(members, topics, mappings) {
+  let nodes = [];
+  let memberNames = {};
+  for (const member of members) {
+    nodes.push({
+      _cssClass: "Member",
+      _labelClass: "memberLabel",
+      name: member.username,
+      id: `member-${member.id}`,
+      submittedAd: member.created_at,
+    });
+    memberNames[member.id] = member.username;
+  }
+
+  for (const topic of topics) {
+    nodes.push({
+      _cssClass: "Skill",
+      _labelClass: "skillLabel",
+      name: topic.name,
+      id: `topic-${topic.id}`,
+      sharerNames: [],
+      learnerNames: [],
+    });
+  }
+
+  const edges = [];
+  const counter = { learners: {}, sharers: {} };
+
+  for (const topic of mappings.filter((item) => item.learner)) {
+    edges.push({
+      type: "learn",
+      _color: "#f1955b",
+      sid: `member-${topic.learner}`,
+      tid: `topic-${topic.topic}`,
+      _svgAttrs: { "stroke-width": "2", opacity: 0.5 },
+    });
+    counter.learners[`topic-${topic.topic}`] = counter.learners[
+      `topic-${topic.topic}`
+    ] || { count: 0, names: [] };
+    counter.learners[`topic-${topic.topic}`].count++;
```

Comment:
we used to keep track of `sharerCount` and `learnerCount` in https://github.com/phivk/gradu.al/blob/1b34d7f61c10962ab7625b0f467305dd39db308c/fetchData.js#L137

I guess you are proposing a slight refactor of the nodes edges data structure?
Please note that `components/MostPopularSkillsSection.vue` is relying on `skill.learnerCount`
see https://github.com/phivk/gradu.al/blob/1b34d7f61c10962ab7625b0f467305dd39db308c/components/MostPopularSkillsSection.vue#L44

Reply:
oh, now I see that you're mapping this below...
I guess it is efficient to do the counting while iterating over the edges, but it makes the function hard to parse for me as a human. Not sure about the best approach here.